packages/block-editor/src/components/block-patterns-list/index.js

```diff
@@ -155,10 +166,16 @@ function BlockPatternList(
 	},
 	ref
 ) {
-	const composite = useCompositeState( { orientation } );
+	const compositeStore = useCompositeStore( { orientation } );
+	const { setActiveId } = compositeStore;
+
+	useEffect( () => {
+		setActiveId( undefined );
+	}, [ setActiveId, shownPatterns, blockPatterns ] );
```

Comment:
Probably a good idea to add an inline comment about what these lines are doing (ie. resetting which composite item is selected any time the list of patterns changes, including when filtering it)